.github/workflows/docker-build-push.yaml

```diff
@@ -7,9 +7,25 @@ on:
     tags:
       - "v*.*.*"
   pull_request:
+  workflow_dispatch:
+  issue_comment:
+    types: [created]
 jobs:
   docker-pipeline:
     runs-on: ubuntu-latest
+    if: |
```

Comment:
Can you add a comment as to why the conditionals are all happening here instead of using the filters in the `on:` block at the top? Perhaps a link to the inspiration for this? How does this approach compare to other well-known OSS projects?

Reply:
Sure.

Spelling it out here quick (since I prefer the soln below if it works) - In short, the `on:` block is necessary but not sufficient to allow these alternate trigger methods. At best the comment-based-trigger is not useful to an outside contributor because like the PR trigger, credentials won't be passed properly. At worst, it's a security risk to allow anyone but contributors and maintainers to trigger the flow - I think secrets become available to the workflow under this context and they can be echoed.

This pattern seems reasonably well-established. Here's the [github doc](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#accessing-secrets) outlining the risk in tokens leaking

Conditional pattern examples:
https://stackoverflow.com/questions/75079250/how-to-trigger-new-job-by-comment-in-pull-request-for-github-actions
https://github.com/orgs/community/discussions/25389